Foundation/NSHTTPCookie.swift

```diff
@@ -327,16 +332,106 @@ public class HTTPCookie : NSObject {
         }
         return ["Cookie": cookieString]
     }
-    
+   
     /// Return an array of cookies parsed from the specified response header fields and URL.
     ///
     /// This method will ignore irrelevant header fields so
     /// you can pass a dictionary containing data other than cookie data.
     /// - Parameter headerFields: The response header fields to check for cookies.
     /// - Parameter URL: The URL that the cookies came from - relevant to how the cookies are interpeted.
     /// - Returns: An array of NSHTTPCookie objects
-    public class func cookies(withResponseHeaderFields headerFields: [String : String], forURL url: URL) -> [HTTPCookie] { NSUnimplemented() }
+    public class func cookies(withResponseHeaderFields headerFields: [String : String], forURL url: URL) -> [HTTPCookie] {
+
+        //HTTP Cookie parsing based on RFC 6265: https://tools.ietf.org/html/rfc6265
+        //Though RFC6265 suggests that multiple cookies cannot be folded into a single Set-Cookie field, this is
+        //pretty common. It also suggests that commas and semicolons among other characters, cannot be a part of 
+        // names and values. This implementation takes care of multiple cookies in the same field, however it doesn't   
+        //support commas and semicolons in names and values(except for dates)
+
+        guard let cookies: String = headerFields["Set-Cookie"]  else { return [] }  
+
+        let nameValuePairs = cookies.components(separatedBy: ";")          //split the name/value and attribute/value pairs
+                                .map({$0.trim()})                           //trim whitespaces
+                                .map({removeCommaFromDate($0)})     //get rid of commas in dates
+                                .flatMap({$0.components(separatedBy: ",")}) //cookie boundaries are marked by commas
+                                .map({$0.trim()})                           //trim again
+                                .filter({$0.caseInsensitiveCompare("HTTPOnly") != .orderedSame})  //we don't use HTTPOnly, do we?
+                                .flatMap({createNameValuePair(pair: $0)})   //create Name and Value properties 
+
+        //mark cookie boundaries in the name-value array
+        var cookieIndices = (0..<nameValuePairs.count).filter({nameValuePairs[$0].hasPrefix("Name")})
+        cookieIndices.append(nameValuePairs.count)  
+
+        //bake the cookies
+        var httpCookies: [HTTPCookie] = []
+        for i in 0..<cookieIndices.count-1 {
+            if let aCookie = createHttpCookie(url: url, pairs: nameValuePairs, start: cookieIndices[i], end: cookieIndices[i+1]) {
+                httpCookies.append(aCookie)
+            }
+        }
     
+        return httpCookies
+    } 
+
+    //Bake a cookie
+    private class func createHttpCookie(url: URL, pairs: [String], start: Int, end: Int) -> HTTPCookie? {
+        var properties: [String:Any] = [:]
+        for index in start..<end {
```

Comment:
Oh I see, we're parsing a subset of the array. I think it might be cleaner to just pass an ArraySlice into this.